airflow/__init__.py

```diff
@@ -42,3 +42,22 @@
 login: Optional[Callable] = None
 
 integrate_plugins()
+
+__all__ = ['__version__', 'login', 'DAG']
+
+
+def __getattr__(name):
+    # PEP-562: Lazy loaded attributes on python modules
+    if name == "DAG":
+        from airflow.models.dag import DAG # pylint: disable=redefined-outer-name
```

Comment:
I think we also need to disable cyclic dependency pylint check as well

Reply:
Worth to try but I have a feeling that this has to be disabled in all places where the cycle occurs.

Reply:
Given this isn't at the top level does pylint still detect it as a cycle?

Reply:
Let's see and disable it if we find we need it. I am not 100% sure -> the algorithm is quite a bit different than python import mechanism (otherwise we would not have the cyclic imports issue at all).